src/Adapter/PhpTerm/PhpTermBackend.php

```diff
@@ -107,6 +109,44 @@ public function clearRegion(ClearType $type): void
         };
     }
 
+    public function cursorPosition(): Position
+    {
+        $this->terminal->queue(Actions::requestCursorPosition());
+        $this->terminal->flush();
+        $start = microtime(true);
+        $this->enableRawMode();
+        $pos = null;
+        while(true) {
+            while (null !== $event = $this->terminal->events()->next()) {
```

Comment:
note this is a bit janky - we will lose any other events that come in during this time, and we should probably have the facility to filter events without removing them

Reply:
it would also be good to use `stream_select` in these cases